services/cloudrun-service/pkg/cmd/server.go

```diff
@@ -46,12 +97,61 @@ func runServer(ctx context.Context) error {
 			Port:     "8443",
 			Opts:     nil,
 			Register: func(srv *grpc.Server) {
-				api.RegisterCloudRunServiceServer(srv, &cloudrun.CloudRunService{})
+				api.RegisterCloudRunServiceServer(srv, &cloudrun.CloudRunService{DBHandler: dbHandler})
+			},
+		},
+		Background: []setup.BackgroundTaskConfig{
+			{
+				Shutdown: nil,
+				Name:     "processDeploymentEvents",
+				Run: func(ctx context.Context, hr *setup.HealthReporter) error {
+					hr.ReportReady("processing deployment events")
+					return processDeploymentEvents(ctx, dbHandler)
+				},
 			},
 		},
-		Background: []setup.BackgroundTaskConfig{},
-		Shutdown:   nil,
+		Shutdown: nil,
 	})
 
 	return nil
 }
+
+func readEnvVar(envName string) (string, error) {
+	envValue, ok := os.LookupEnv(envName)
+	if !ok {
+		return "", fmt.Errorf("could not read environment variable '%s'", envName)
+	}
+	return envValue, nil
+}
+
+func processDeploymentEvents(ctx context.Context, dbHandler *db.DBHandler) error {
+	log := logger.FromContext(ctx).Sugar()
+	for {
+		time.Sleep(5 * time.Second)
+		queuedDeployments, err := dbx.GetQueuedDeployments(ctx, dbHandler)
+		if err != nil {
+			log.Errorf("failed to read queued deployment events: %v", err)
+			continue
+		}
+		if len(queuedDeployments) == 0 {
+			log.Info("no queued deployments to process")
+			continue
+		}
```

Comment:
```suggestion
		queuedDeployments, err := dbx.GetQueuedDeployments(ctx, dbHandler)
		if err != nil {
			log.Errorf("failed to read queued deployment events: %v", err)
			continue
		}
		if len(queuedDeployments) == 0 {
			log.Info("no queued deployments to process")
        		time.Sleep(5 * time.Second)
			continue
		}
```

Reply:
We should not sleep when there is work to do

Reply:
I'd rather keep the sleep at the beginning of the loop so that if there is a case where `GetQueuedDeployments` always returns an error the loop waits before making another call to the database. Otherwise, loop would continuously poll the database without any delays